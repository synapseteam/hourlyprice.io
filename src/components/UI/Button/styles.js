import { css } from "@emotion/react";
import { getStyle } from "utils/generic";

const green = "#174f5e";
const lightgreen = "#5cbabc";
const white = "#ffffff";
const lightGrey = "#e5e5e5";
const black = "#000000";
const darkGrey = "#d5d3d3";

export const styles = {
  buttonContainer: css`
    width: 100%;
    margin: 2rem 0;
  `,
  button: css`
    border: 1px solid ${lightgreen};
    border-radius: 0.5rem;
    background-color: ${green};
    width: 100%;
    color: ${white};
    padding: 0.8rem;
    font-weight: 600;
    cursor: pointer;

    &:hover {
      background-color: ${lightgreen};
    }
  `,

  buttonLight: css`
    border: none;
    background-color: ${lightGrey};
    color: ${black};

    &:hover {
      background-color: ${darkGrey};
    }
  `,
  getStyle,
};
