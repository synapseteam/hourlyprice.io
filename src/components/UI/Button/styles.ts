import { css, Theme } from "@emotion/react";

export const styles = {
  buttonContainer: css`
    width: 100%;
    margin: 1.2rem 0 0.4rem;
  `,
  button: (theme: Theme) => css`
    border: ${theme.btnBorder};
    border-radius: 0.4rem;
    background-color: ${theme.nonary};
    width: 100%;
    color: ${theme.denary};
    padding: 0.7rem;
    font-weight: 600;
    cursor: pointer;

    &:hover {
      background-color: ${theme.btnHoverColor};
    }
    &:disabled {
      background-color: ${theme.btnDisabledBgColor};
      color: ${theme.btnDisabledColor};
      cursor: not-allowed;
    }
  `,
};
