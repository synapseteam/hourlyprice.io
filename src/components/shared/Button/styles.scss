$green: #174f5e;
$lightgreen: #5cbabc;
$mediumGreen: #2d8587;
$white: #ffffff;
$light-grey: #e5e5e5;
$black: #000000;

.calculate-btn-container {
  width: 100%;
  margin: 2rem 0;
  .calculate-btn {
    border: 1px solid $lightgreen;
    border-radius: 0.5rem;
    background-color: $green;
    width: 100%;
    color: $white;
    padding: 0.8rem;
    font-weight: 600;
    cursor: pointer;

    &:hover {
      background-color: $lightgreen;
    }
  }
}

.light-btn {
  .calculate-btn {
    background-color: $light-grey;
    border: none;
    color: $black;

    &:hover {
      background-color: darken($light-grey, 10);
    }
  }
}
