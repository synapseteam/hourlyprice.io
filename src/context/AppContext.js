import { createContext, useReducer, useContext } from "react";

const AppThemeContext = createContext();

const initState = {
  darkMode: true,
};

const TOGGLE_THEME = "TOGGLE_THEME";

function reducer(state, action) {
  switch (action.type) {
    case TOGGLE_THEME:
      return {
        ...state,
        darkMode: !state.darkMode,
      };
    default:
      throw Error("impossible case while switching Theme");
  }
}

export const AppThemeProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initState);
  const value = [state, dispatch];

  return (
    <AppThemeContext.Provider value={value}>
      {children}
    </AppThemeContext.Provider>
  );
};

export const useAppThemeContext = () => {
  const context = useContext(AppThemeContext);

  if (!context) {
    throw Error("useAppThemeContext should be used within AppThemeProvider");
  }

  return context;
};

export const toggleTheme = () => {
  return { type: TOGGLE_THEME, payload: null };
};

export default AppThemeContext;
