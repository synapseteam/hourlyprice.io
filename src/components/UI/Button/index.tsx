/** @jsxImportSource @emotion/react */
import { SerializedStyles, Theme } from "@emotion/react";
import { styles } from "./styles";

interface Props {
  type?: "button" | "reset" | "submit";
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  children: JSX.Element[] | JSX.Element | string;
  disabled?: boolean;
  classname?: SerializedStyles;
  classnameContainer?: SerializedStyles;
  form?: string;
}

const Button: React.FC<Props> = ({
  type = "button",
  onClick,
  children,
  disabled = false,
  classname,
  classnameContainer,
  form,
}): JSX.Element => {
  return (
    <div css={[styles.buttonContainer, classnameContainer]}>
      <button
        css={[styles.button, classname]}
        type={type}
        color="red"
        onClick={onClick}
        disabled={disabled}
        form={form}
      >
        {children}
      </button>
    </div>
  );
};

export default Button;
