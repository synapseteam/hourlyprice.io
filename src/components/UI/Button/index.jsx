/** @jsxImportSource @emotion/react */

import React from "react";
import PropTypes from "prop-types";
import { useAppThemeContext } from "context/AppContext";
import { useCustomTranslation } from "i18n";

import { styles } from "./styles";

export default function ButtonSubmit() {
  const [{ darkMode }] = useAppThemeContext();
  const [t] = useCustomTranslation();

  return (
    <div
      css={() => styles.getStyle(darkMode, "buttonContainer")}
      className="button__container"
    >
      <button
        css={() => styles.getStyle(darkMode, "button")}
        type="submit"
        className="button"
        color="red"
      >
        {t("btnResult").toUpperCase()}
      </button>
    </div>
  );
}

ButtonSubmit.propTypes = {
  formId: PropTypes.string,
};
