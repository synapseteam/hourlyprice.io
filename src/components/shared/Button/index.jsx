import React from "react";
import PropTypes from "prop-types";

import { useAppThemeContext } from "context/AppContext";
import { useCustomTranslation } from "i18n";

import "components/shared/Button/styles.scss";

export default function ButtonSubmit({ formId }) {
  const [context] = useAppThemeContext();
  const [t] = useCustomTranslation();

  const additionalProps = formId
    ? {
        form: formId,
      }
    : null;

  return (
    <div
      className={
        context.darkMode
          ? "calculate-btn-container"
          : "calculate-btn-container light-btn"
      }
    >
      <button className="calculate-btn" {...additionalProps}>
        {t("btnResult").toUpperCase()}
      </button>
    </div>
  );
}

ButtonSubmit.propTypes = {
  formId: PropTypes.string,
};
