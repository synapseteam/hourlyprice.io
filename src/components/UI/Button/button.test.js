import React from "react";
import renderer from "react-test-renderer";

import { AppThemeProvider } from "context/AppContext";
import Button from "components/UI/Button";

if (!global.React) {
  global.React = React;
}

test("Button rendered correctly, snapshot matched", () => {
  const context = {
    darkMode: true,
  };
  const value = [context];
  const tree = renderer
    .create(
      <AppThemeProvider value={value}>
        <Button />
      </AppThemeProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
