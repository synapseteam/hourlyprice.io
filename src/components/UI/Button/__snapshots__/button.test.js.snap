// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendered correctly, snapshot matched 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-15lh1gu-buttonContainer{width:100%;margin:2rem 0;}",
      }
    }
    data-emotion="css 15lh1gu-buttonContainer"
  />,
  <div
    className="button__container css-15lh1gu-buttonContainer"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-lcahke-button{border:1px solid #5cbabc;border-radius:0.5rem;background-color:#174f5e;width:100%;color:#ffffff;padding:0.8rem;font-weight:600;cursor:pointer;}.css-lcahke-button:hover{background-color:#5cbabc;}",
        }
      }
      data-emotion="css lcahke-button"
    />
    <button
      className="button css-lcahke-button"
      color="red"
      type="submit"
    >
      RESULT
    </button>
  </div>,
]
`;
